DOC: Add user documentation for tegra support (#70)
conda-forge/cuda-feedstock: recipe/doc/recipe_guide.md

This guide assumes that you already know how to write and compile code that supports such behavior.
If your library is properly configured as such, you will need to do a bit of extra work to ensure that your conda package supports this as well:

> - You must [ignore_run_exports_from`](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#export-runtime-requirements) of any CUDA library packages that your package depends on. Otherwise, the `run_exports` would require the runtime libraries to have versions equal to or greater than the versions used to build the package.

Reviewer: Ignoring run exports is just needed for `-dev` packages, right?
```suggestion
- You must define [`ignore_run_exports_from`](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#export-runtime-requirements) for any CUDA library `-dev` packages that your package depends on. Otherwise, the `run_exports` would require the runtime libraries to have versions equal to or greater than the versions used to build the package.
```

x86-64 Linux) -- and the `cuda-nvcc_${TARGET_PLATFORM}` packages -- each of which is architecture-independent and may be installed on any target, but are only suitable for use in compiling code for the specified target platform.
This approach allows using host machines with a single platform to compile code for multiple platforms.

## Building for ARM Tegra devices

> [!IMPORTANT]
> Support for Tegra builds on conda-forge is only available for CUDA 12.9 and later.

The [arm-variant](https://github.com/conda-forge/arm-variant-feedstock) package allows
end-users to select which variant of ARM packages are installed into their environment.
However, since there are no Tegra devices available for compilation, these packages must be
cross-compiled from x86. Recipes that wish to build for both SBSA ARM and Tegra ARM devices
should have something like the following in their recipe:

```yaml
requirements:
  build:
    - {{ compiler('cuda') }}

>     - arm-variant * {{ arm_variant_type }}

Author: Yes. This is fixed.

  host:

>     - arm-variant * {{ arm_variant_type }}

Author: ```suggestion
requirements:
  build:
    - {{ compiler('cuda') }}
    - arm-variant * {{ arm_variant_type }}  # [linux and aarch64]
```
Since I first drafted these docs, we have added strong run_exports to `arm-variant`, so it should be sufficient to just add `arm-variant` to the build environment. Then the host and run environments will also be constrained to the same `arm-variant`.

```

> [!NOTE]
> The `arm-variant` package will cause overlinking warnings for itself from `conda-build`.
> This is expected and may be safely ignored.

where the `recipe/conda_build_config.yaml` contains something like:

```yaml
arm_variant_type:  # [linux and aarch64]
  - sbsa           # [linux and aarch64]
  - tegra          # [linux and aarch64]
```

where the `conda-forge.yml` contains something like:

```yaml
build_platform:
  linux_aarch64: linux_64
provider:
  linux_aarch64: default
```

The compute capabilities for GPUs on Tegra and non-Tegra devices are mutually exclusive, and
device-code compiled for Tegra and non-Tegra devices are not interchangeable, so one build
cannot service all ARM variants. For developer convenience, the CUDA compiler package
activation script for `cuda-nvcc >=12.8` automatically sets the environment variables
`CUDAARCHS` and `TORCH_CUDA_ARCH_LIST` to all supported CUDA architectures for the target

> platform. These environment variables are consumed by CMake and PyTorch respectively.

Author: ```suggestion
The compute capabilities for GPUs on Tegra and non-Tegra devices are mutually exclusive, and
device-code compiled for Tegra and non-Tegra devices are not interchangeable, so one build
cannot service all ARM variants. For developer convenience, the CUDA compiler package
activation script for `cuda-nvcc >=12.8` automatically sets the environment variable
`CUDAARCHS` to all supported CUDA architectures for the target platform when `CONDA_BUILD` is
detected and when `CUDAARCHS` is not already set. This environment variable is a standard CMake
environment variable.
```
